### ISSUE TITLE
Investigate updating carbon-cycle timestep

### ISSUE THREAD

**Issue by cahartin** [JGCRI/hector#195]
Impulse response different with a smaller carbon-cycle solver @akschw04.  @akschw04 can you add details on this?

**Comment by akschw04** [comment]
In order for Hector-deoclime to successfully solve after a large [CO2] perturbation or step, I had to alter two timesteps: (1) the timestep in carbon-cycle-solver.cpp from dt(0.3) to dt(0.1) and (2) the ocean_component.hpp OCEAN_MIN_TIMESTEP from 0.3 to 0.01. 

Changing these two timesteps also improved numerical noise in the atmosphere-ocean carbon flux response from various emissions perturbations in 2010, seen in the figure here. 

![image](https://user-images.githubusercontent.com/9463017/27976607-480a34fc-6335-11e7-96f6-48c172228322.png)

**Comment by rplzzz** [comment]
Your problem isn't the size of your time steps.  Your problem is that you've hard-wired the size of your time steps, and you're not estimating and controlling the error in the integration.

**Comment by cahartin** [comment]
@rplzzz it's currently hardcoded in Hector and Adria made the time step shorter. It would be useful to have Hector some how figure this out on it's own...

**Comment by rplzzz** [comment]
That's what I was getting at.  I suspect this is a consequence of the separation between our components.  We can't easily do an adaptive time step because the integrations are being done in separate components.  I sketched out a way to do this a long time ago, but it involves some tricky programming in order to do it and still keep the components independent.

The next best thing would be to monitor the error in the integration and raise a warning if it gets outside of prescribed bounds.

**Comment by bpbond** [comment]
>We can't easily do an adaptive time step because the integrations are being done in separate components

Right. The current method works around this, kind of, but in a duct-tape-and-string kind of way.
